Mittons.Fixtures/Docker/Gateways/IContainerGateway.cs

```diff
@@ -0,0 +1,26 @@
+using System.Collections.Generic;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Mittons.Fixtures.Docker.Gateways
+{
+    public interface IContainerGateway
+    {
+        Task<string> RunAsync(string imageName, string command, IEnumerable<KeyValuePair<string, string>> options, CancellationToken cancellationToken);
+
+        Task RemoveAsync(string containerId, CancellationToken cancellationToken);
+
+        Task<IPAddress> GetDefaultNetworkIpAddressAsync(string containerId, CancellationToken cancellationToken);
```

Comment:
I was curious what the significance is of `Default` in the name here. Is this something you'd consider in the same context, or is this naming still inline what's intended? Default vs. something else?

`GetDefaultNetworkIpAddressAsync` vs `GetNetworkIpAddressAsync`

Reply:
In this context, "default" means the default docker network the container is added to when first created, so it has meaning compared to every other network a container might later be connected to.

This is important in environments like running in an ADO docker environment, where the host can _only_ talk to the container across the default network.

Does that make sense?

Reply:
👍 makes sense